src/components/CommenAdder.jsx

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect, } from "react";
+import { getUsers, postComment } from "../utils/api";
+import { useParams } from "react-router-dom";
+
+
+
+function CommentAdder (setComments) {
+    const {id} = useParams();
+    const [users, setUsers] = useState([]);
+    const [addComment, setAddComment] = useState("");
+    const [selectedUser, setSelectedUser] = useState("");
+
+
+    useEffect(() => {
+        getUsers()
+        .then((response) => {
+            setUsers(response.users)
+        })
+        .catch((err) => {
+            console.log(err)
+        })
+    }, []);
+
+    function handleAddComment(event) {
+        setAddComment(event.target.value);
+      }
+
+      function handleUserChange(event) {
+        setSelectedUser(event.target.value);
+      }
+
+    
+      function handleSubmit(event) {
+        event.preventDefault();
+    
+        postComment(id, selectedUser, addComment)
+          .then((response) => {
+            setComments((currentComments) => {
+                return [response, ...currentComments]
+            })
+        })
+        .catch((err) => {
+            console.log(err)
```

Comment:
Currently if the post request doesn't work (for example if the internet goes out) you `console.log` the error, it would  be good if you also told the user via an error component as well 🙂